core/src/main/java/software/amazon/awssdk/http/pipeline/stages/ApplyUserAgentStage.java

```diff
@@ -31,20 +32,38 @@
 public class ApplyUserAgentStage implements MutableRequestToRequestPipeline {
 
     private final LegacyClientConfiguration config;
+    private final String clientName;
+    private final String clientType;
 
-    public ApplyUserAgentStage(HttpClientDependencies dependencies) {
+    public ApplyUserAgentStage(HttpClientDependencies dependencies,
+                               String clientName,
+                               String clientType) {
```

Comment:
I don't think it's worth the complexity. We will choose good names for our clients and it's up to others to use good names for theirs. Another idea is have the implementation append it that way they have control whether is reported on or not (to guard against the potentially crappy client name)